### DISCUSSION TITLE
could I steer somehow the volume from a weblink

### DISCUSSION THREAD

**Discussion by Mannshoch** [MiczFlor/RPi-Jukebox-RFID#2347] [Q&A]
I use http shortcuts for some other smarthome cases.
Would be nice if I could increase or decrease the volume by link.
localhost/volume/+, or /volume/75%

Is there a solution for that?

**Comment by Mannshoch** [comment]
Any Ideas?
could I somehow use volume.php for that?

**Comment by s-martin** [comment]
V2 has MQTT support (https://github.com/MiczFlor/RPi-Jukebox-RFID/tree/develop/components/smart-home-automation/MQTT-protocol), which should support your use case.

For V3 there is no MQTT support available yet, people interested in implementing or supporting this feature should coordinate in #1963

**Reply by Mannshoch** [reply]
Sounds promising:
I followed the advices from the link but I get this if I try to start MQTT.
```

/home/pi/RPi-Jukebox-RFID/scripts/daemon_mqtt_client.py
/home/pi/RPi-Jukebox-RFID/scripts/daemon_mqtt_client.py:594: DeprecationWarning: Callback API version 1 is deprecated, update to latest version
  client = mqtt.Client(callback_api_version=mqtt_enum.CallbackAPIVersion.VERSION1, client_id=config.get("mqttClientId"))
Traceback (most recent call last):
  File "/home/pi/RPi-Jukebox-RFID/scripts/daemon_mqtt_client.py", line 604, in <module>
    client.tls_set(
  File "/usr/local/lib/python3.9/dist-packages/paho/mqtt/client.py", line 1285, in tls_set
    context.load_cert_chain(certfile, keyfile, keyfile_password)
FileNotFoundError: [Errno 2] No such file or directory
```
Any way. I do not really understand this MQTT.
On my  Smartphone I use HTTP shortcut to send command with curl. How whould that work with MQTT?

**Reply by Mannshoch** [reply]
Weil searching MQTT i crossed the line with webhook

my `hook.json`
```
[
  {
    "id": "mein-webhook",
    "execute-command": "/home/pi/RPi-Jukebox-RFID/scripts/playout_controls.sh",
    "command-working-directory": "/home/pi/RPi-Jukebox-RFID/scripts/",
    "pass-arguments-to-command":
    [
      {
        "source": "string",
        "name": "-c=setvolume"
      },
      {
        "source": "payload",
        "name": "wert"
      }
    ],
    "trigger-rule":
    {
      "and":
      [
        {
          "match":
          {
            "type": "regex",
            "regex": "^-v=(0|[1-9][0-9]?|100)$",
            "parameter":
            {
              "source": "payload",
              "name": "wert"
            }
          }
        }
      ]
    }
  }
]
```

and this curl
`curl -X POST -H "Content-Type: application/json" -d '{"wert": "-v=80"}' http://<IP>:9000/hooks/mein-webhook`

With the app [HTTP Request Shortcuts](https://f-droid.org/de/packages/ch.rmy.android.http_shortcuts/) I'm now able to set the volume from my mobile phone
